### PR TITLE
Test-runner can filter tests by name

### PR DESCRIPTION
**Pull request by aardappel**

### CODE REVIEW

File: scripts/test/shared.py [hunk 1]
```diff
@@ -362,6 +366,8 @@ def get_tests(test_dir, extensions=[]):
         tests += glob.glob(os.path.join(test_dir, '*'))
     for ext in extensions:
         tests += glob.glob(os.path.join(test_dir, '*' + ext))
+    if options.test_name_filter:
+        tests = list(filter(lambda n: n.find(options.test_name_filter) >= 0, tests))
```
Comment (sbc100) [inline scripts/test/shared.py:370]: Can you use list comprehension instead here?

```
tests = [t for t in tests if t.find(options.test_name_filter) >= 0]
```

Even if you don't use list comprehension isn't the `list()` on the outside redundant?

Comment (aardappel) [reply scripts/test/shared.py:370]: `filter` returns an iterator, not a list, apparently.

Comment (aardappel) [reply scripts/test/shared.py:370]: not going to open a new PR to change it into a comprehension if you don't mind..

Comment (sbc100) [reply scripts/test/shared.py:370]: No worries, my feedback crossed paths with landing.

**Pull request merged**